cpp/sophus/image/CMakeLists.txt

```diff
@@ -21,23 +21,23 @@ foreach(x ${sophus_image_src_prefixes} )
     list(APPEND sophus_image_h ${x}.h)
 endforeach()
 
-add_library(sophus_image SHARED
+farm_ng_add_library(sophus_image
+  NAMESPACE sophus
+  INCLUDE_DIR ../../
+  HEADERS
+  ${sophus_image_h}
+  SOURCES
   ${sophus_image_cpp}
-  ${sophus_image_h})
+)
 target_link_libraries(sophus_image sophus_geometry)
-add_library (sophus::sophus_image ALIAS sophus_image)
 
-target_sources(sophus_image INTERFACE
-               FILE_SET HEADERS
-               BASE_DIRS ../..
-               FILES ${sophus_image_h})
 
 
 if(${BUILD_SOPHUS_TESTS})
   foreach(test_basename ${sophus_image_src_prefixes})
     farm_ng_add_test(${test_basename}
                           PARENT_LIBRARY sophus_image
-                          LINK_LIBRARIES
+                          LINK_LIBRARIES sophus_image
```

Comment:
why is that needed? should'nt we always link against the parent lib?

Reply:
For some reason the macro didn't link against the parent lib, which is used just for the test name. We can make it default in a follow up.